File: Progressio_backend/Progressio.Services/Database/Entities/Content.cs
namespace Progressio.Services.Database.Entities;

public class Content
{
    public int Id { get; set; }
    public string Title { get; set; } = null!;
    public string? Description { get; set; }
    public string? CoverImageUrl { get; set; }
    public int ContentTypeId { get; set; }
    public int? AgeRatingId { get; set; }
    public int? LanguageId { get; set; }
    public int? ReleaseYear { get; set; }
    public double AvgRating { get; set; }
    public int TotalRatings { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ContentType ContentType { get; set; } = null!;
    public AgeRating? AgeRating { get; set; }
    public Language? Language { get; set; }
    public ICollection<ContentGenre> ContentGenres { get; set; } = [];
    public ICollection<ContentPlatform> ContentPlatforms { get; set; } = [];
    public ICollection<Season> Seasons { get; set; } = [];
    public ICollection<Chapter> Chapters { get; set; } = [];
    public ICollection<Character> Characters { get; set; } = [];
    public ICollection<UserContentProgress> UserProgresses { get; set; } = [];
    public ICollection<Review> Reviews { get; set; } = [];
    public ICollection<ContentComment> Comments { get; set; } = [];
    public ICollection<RecommendationLog> RecommendationLogs { get; set; } = [];
}

File: ui/progressio_mobile/lib/model/content.dart
class Content {
  final int id;
  final String title;
  final String? description;
  final String? coverImageUrl;
  final int contentTypeId;
  final String? contentTypeName;
  final int? ageRatingId;
  final String? ageRatingName;
  final int? languageId;
  final String? languageName;
  final int? releaseYear;
  final double avgRating;
  final int totalRatings;
  final bool isActive;
  final DateTime createdAt;
  final List<String> genres;

  Content({
    this.id = 0,
    this.title = '',
    this.description,
    this.coverImageUrl,
    this.contentTypeId = 0,
    this.contentTypeName,
    this.ageRatingId,
    this.ageRatingName,
    this.languageId,
    this.languageName,
    this.releaseYear,
    this.avgRating = 0,
    this.totalRatings = 0,
    this.isActive = true,
    DateTime? createdAt,
    List<String>? genres,
  })  : createdAt = createdAt ?? DateTime.now(),
        genres = genres ?? [];

  factory Content.fromJson(Map<String, dynamic> json) {
    return Content(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      description: json['description'],
      coverImageUrl: json['coverImageUrl'],
      contentTypeId: json['contentTypeId'] ?? 0,
      contentTypeName: json['contentTypeName'],
      ageRatingId: json['ageRatingId'],
      ageRatingName: json['ageRatingName'],
      languageId: json['languageId'],
      languageName: json['languageName'],
      releaseYear: json['releaseYear'],
      avgRating: (json['avgRating'] ?? 0).toDouble(),
      totalRatings: json['totalRatings'] ?? 0,
      isActive: json['isActive'] ?? true,
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
      genres: json['genres'] != null
          ? List<String>.from(json['genres'])
          : [],
    );
  }
}